### PR TITLE
Feature/hotspot map

### PR DESCRIPTION
**Pull request by bryzettler**
- Migrates our mapping logic to [maplibre-react-native](https://github.com/maplibre/maplibre-react-native) with protomaps.
- Adds a new hotspot map view and a inner drawer that replaces the HotspotDetailScreen
- Update assert location to match new ui for map

https://github.com/helium/wallet-app/assets/694011/40891899-6f0b-4a6a-9cc4-f72bd29f413a

ui

### CODE REVIEW

File: src/hooks/useKeyToAsset.ts [hunk 1]
```diff
@@ -1,8 +1,11 @@
+import { IdlAccounts } from '@coral-xyz/anchor'
 import { useAnchorAccount } from '@helium/helium-react-hooks'
 import { HeliumEntityManager } from '@helium/idls/lib/types/helium_entity_manager'
 import { usePublicKey } from './usePublicKey'
 
 const type = 'keyToAssetV0'
+export type KeyToAssetV0 = IdlAccounts<HeliumEntityManager>['keyToAssetV0']
```
Comment (ChewingGlass) [inline src/hooks/useKeyToAsset.ts:7]: Weird that this is exported from here but not used here?

File: src/utils/solanaUtils.ts [hunk 2]
```diff
@@ -1091,6 +1115,94 @@ export async function getCachedKeyToAssets(
...
+    await cache.searchMultiple(
+      infoKeys,
+      (pubkey, account) => ({
+        pubkey,
+        account,
+        info: hemProgram.coder.accounts.decode<
+          IdlAccounts<HeliumEntityManager>['iotHotspotInfoV0']
+        >('IotHotspotInfoV0', account.data),
+      }),
+      true,
```
Comment (ChewingGlass) [inline src/utils/solanaUtils.ts:1154]: This should be false, as iot infos are not static (they can change, for example in an assert location). Setting this to true will tell the cache to not watch for changes, which could cause issues updating.

File: src/utils/solanaUtils.ts [hunk 3]
```diff
@@ -1091,6 +1115,94 @@ export async function getCachedKeyToAssets(
...
+    await cache.searchMultiple(
+      infoKeys,
+      (pubkey, account) => ({
+        pubkey,
+        account,
+        info: hemProgram.coder.accounts.decode<
+          IdlAccounts<HeliumEntityManager>['mobileHotspotInfoV0']
+        >('MobileHotspotInfoV0', account.data),
+      }),
+      true,
```
Comment (ChewingGlass) [inline src/utils/solanaUtils.ts:1198]: Same here

File: ios/HeliumWallet.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -471,6 +471,9 @@
 				00E356F51AD99517003FC87E /* PBXTargetDependency */,
 			);
 			name = HeliumWalletTests;
+			packageProductDependencies = (
```
[the remark this answers is not in the document]
Comment (bryzettler) [reply ios/HeliumWallet.xcodeproj/project.pbxproj:474]: I think this comes from maplibre-react-native and the fact of this

> This project originated as a fork of [rnmapbox](https://github.com/rnmapbox/maps), a community-maintained React Native Library for building maps with the Mapbox iOS and Android mobile SDKs. The original product supported both Mapbox and MapLibre for some time, but as the MapLibre and Mapbox SDKs have diverged, it has become necessary to separate the projects into specific wrappers by underlying renderer.

File: src/hooks/useKeyToAsset.ts (hunk 1 above)
Comment (bryzettler) [reply src/hooks/useKeyToAsset.ts:7]: I mirrored what was going on with `useMobileInfo` and `useIotInfo`

**Comment by tyler-whitman** [comment]
Once you merge this we will have to update Bitrise to set up the new env vars. I can run you guys through how to do that, let me know when would work for you.

File: src/store/slices/hotspotsSlice.ts [hunk 5]
```diff
@@ -25,86 +27,81 @@ const initialState: HotspotsByCluster = {
...
-
-    const hotspotsWithMetadata = await solUtils.getCompressedNFTMetadata(
-      fetchedHotspots.items,
+      1000,
     )
 
-    const hotspotsWithPendingRewards =
-      await solUtils.annotateWithPendingRewards(
+    let hotspots: CompressedNFT[] = []
+    while (hotspots.length < fetchedHotspots.total) {
```
Comment (ChewingGlass) [inline src/store/slices/hotspotsSlice.ts:51]: This is the spot that's causing a second searchAssets fetch. You can use fetchedHotspots.grandTotal to figure out if you need to fetch more

File: src/store/slices/hotspotsSlice.ts [hunk 6]
```diff
@@ -1,19 +1,21 @@
...
-  hotspotsWithMeta: HotspotWithPendingRewards[]
   loading: boolean
   fetchingMore: boolean
   onEndReached: boolean
   page: number
   totalHotspots?: number
+
+  hotspotsById: Record<string, CompressedNFT>
+  hotspotsMetadataById: Record<string, any>
+  hotspotsRewardsById: Record<string, any>
```
Comment (ChewingGlass) [inline src/store/slices/hotspotsSlice.ts:18]: can this have a type that's not `any`?

**Review by ChewingGlass — CHANGES_REQUESTED** [review]

**Review by ChewingGlass — APPROVED** [review]

**Pull request merged**